test/test_e2e.rb

```diff
@@ -104,4 +109,27 @@ def test_spec_file_splitting
       "./spec/fast_spec.rb",
     ], queue)
   end
+
+  def test_suite_with_failures_and_fail_fast
+    queue = exec_build("failing_suite", "--fail-fast 1")
+
+    assert_equal 1, queue.fail_fast
+    assert_operator queue.unprocessed_jobs.length, :<=, 3
+    assert_operator queue.processed_jobs.length, :<=, 2
+    assert queue.build_failed_fast?
+    refute queue.build_successful?
+    assert_equal queue.fail_fast, queue.example_failures.length +
```

Comment:
This is essentially what fail_fast checks. IMO it should be inside a unit test, but since we do not have those, I thought of validating that the state has changed the way it should

Reply:
Ah I see. SGTM!